mantle/kola/harness.go

```diff
@@ -272,6 +273,73 @@ func testRequiresInternet(test *register.Test) bool {
 	return false
 }
 
+type DenyListObj struct {
+	Pattern string   `yaml:"pattern"`
+	Tracker string   `yaml:"tracker"`
+	Streams []string `yaml:"streams"`
+	Arches  []string `yaml:"arches"`
+}
+
+type ManifestData struct {
+	AddCommitMetadata struct {
+		FcosStream string `yaml:"fedora-coreos.stream"`
+	} `yaml:"add-commit-metadata"`
+}
+
+func parseDenyListYaml() {
+	var objs []DenyListObj
+
+	// Parse kola-denylist into structs
+	denyListFile, err := ioutil.ReadFile("src/config/kola-denylist.yaml")
+
+	if err != nil {
+		return
+	}
```

Comment:
We probably want to behave differently based on what type of error we hit. If the file doesn't exist (it's not required so that's OK) then we want to behave differently than if the file exists and we can't open it for whatever reason.

Reply:
Makes sense. My assumption would be that we would apply the same error checking logic to `manifest.yaml`.

Reply:
I guess that depends. Is `manifest.yaml` optional here? I'm thinking it's probably required but maybe I'm wrong.

If it's required then we can just throw any error back at the caller. If it's optional then we'd need to treat it like `kola-denylist.yaml`, yeah.

Reply:
If we are trying to replicate the same functionality that was originally implemented in `cmd-kola`, `FileNotFoundError`s were caught for both `manifest.yaml` and `kola-denylist.yaml` ([link](https://github.com/coreos/coreos-assembler/blob/main/src/cmd-kola#L61)).

Reply:
ok so we can consider them both optional 👍